src/test/regress/citus_tests/run_test.py

```diff
@@ -54,13 +55,37 @@
 use_base_schedule = args["use_base_schedule"]
 use_whole_schedule_line = args["use_whole_schedule_line"]
 
-test_files_to_skip = [
-    "multi_cluster_management",
-    "multi_extension",
-    "multi_test_helpers",
-    "multi_insert_select",
-]
-test_files_to_run_without_schedule = ["single_node_enterprise"]
+
+class TestDeps:
+    schedule: str | None
+    direct_extra_tests: list[str]
+
+    def __init__(self, schedule, extra_tests=None, repeatable=True):
+        self.schedule = schedule
+        self.direct_extra_tests = extra_tests or []
+        self.repeatable = repeatable
+
+    def extra_tests(self):
+        all_deps = OrderedDict()
+        for direct_dep in self.direct_extra_tests:
+            if direct_dep in deps:
+                for indirect_dep in deps[direct_dep].extra_tests():
+                    all_deps[indirect_dep] = True
```

Comment:
can we have a recursive function for this?
maybe not in this pr but we should definitely consider creating an issue for that for the future

Reply:
In what way is the current implementation not recursive?

Reply:
ah, my bad, misread.